### PR TITLE
PAXURL-323: embed dependencies

### PR DESCRIPTION
**Pull request by bastijns-jeroen**

### CODE REVIEW

File: pax-url-assembly/pom.xml [hunk 1]
```diff
@@ -29,6 +29,34 @@
         <groupId>org.apache.felix</groupId>
         <artifactId>maven-bundle-plugin</artifactId>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
```
Comment (ANierbeck) [inline pax-url-assembly/pom.xml]: are you sure this plugin is needed?
Cause with your change above, using the Embed-Dependency it those org.ops4j.io and lang packages are already included. So the shading isn't needed any longer. Unless you want an extra Bundle that contains those, then you need to make sure those packages aren't included through the bnd file syntax.

Comment (gnodet) [inline pax-url-assembly/pom.xml]: Maybe this is for the sources jar ?
It has always been quite tricky to use embed packages using the maven bundle plugin and generate a source jar containing those sources...

Comment (ANierbeck) [inline pax-url-assembly/pom.xml]: Ahh, yes, that could be interesting, though still would be nice to get feedback from the author :-)

Comment (bastijns-jeroen) [inline pax-url-assembly/pom.xml]: @ANierbeck 
I initially had an issue with org.ops4j.io and org.ops4j.lang not being on the classpath. Embedding should fix this, but like you mentioned, I should exclude them from the the export-packages in the bnd file. As I don't want to make them available for others. Shading should be no longer necessary.

@gnodet I didn't found a Maven Source Plugin to generate sources...

Comment (bastijns-jeroen) [inline pax-url-assembly/pom.xml]: @ANierbeck 
Can you tell me why pax-url-aether uses the same strategy? There the dependencies on org.ops4j.lang etc are also embedded and shaded.

Comment (ANierbeck) [inline pax-url-assembly/pom.xml]: afaik, aether is a special case due to maven itself. Where other maven packages might be available already. 
https://ops4j1.jira.com/browse/PAXURL-251
I'm still unsure that this is actually needed. As all other modules don't depend on shading for including those embedded packages. This is usually done by the maven-bundle-plugin. Aether on the other hand might also be used outside the osgi context.

Comment (bastijns-jeroen) [inline pax-url-assembly/pom.xml]: Ok, I will remove the shading as it is not necessary for me. And just embed the dependencies.

**Comment by bastijns-jeroen** [comment]
Updated the pull request and the issue (https://ops4j1.jira.com/browse/PAXURL-323).

**Comment by ANierbeck** [comment]
cool, feel free to merge this pr :)

**Comment by bastijns-jeroen** [comment]
Thx! Any idea when you could release a version with this pull request included? I have another issue on pax-runner for updating pax-url to the latest version.

**Comment by ANierbeck** [comment]
we can do a release anytime, btw. if you have an oss.sonatype account, you can do this release too

**Comment by bastijns-jeroen** [comment]
Nice, I will create an account and release it.

**Comment by ANierbeck** [comment]
documentation about the release process: 
https://ops4j1.jira.com/wiki/display/ops4j/Releasing

**Comment by bastijns-jeroen** [comment]
@ANierbeck Released 2.4.5 but can't complete the releasing procedure. I haven't got the permissions to change the status of Jira tickets. I have to add the release version to this one and mark it as completed.

**Comment by ANierbeck** [comment]
@bastijns-jeroen I added some more groups to your user, now you should be able to close it. 
And don't forget to announce this release on the ops4j mailinglist: 
- https://groups.google.com/forum/#!forum/ops4j
- https://groups.google.com/forum/#!forum/ops4j-announcement

thanks, Achim 
And happy Holidays

**Comment by bastijns-jeroen** [comment]
@ANierbeck Closed the two Jira-tickets but don't have the permission to mark 2.4.5 as released.
Announced the release on ops4j but don't have the permissions for posting on ops4j-announcement. Made a Confluence blog entry to announce the release.

Thanks for the help! Happy Holidays!

**Comment by jbonofre** [comment]
Let me do it for you.

**Comment by jbonofre** [comment]
Done

**Comment by bastijns-jeroen** [comment]
Thx!

**Comment by CodingFabian** [comment]
hi @bastijns-jeroen is there any specific reason you only did this for assembly? for example I am using url-aether which for me looks like it is incorrectly shading too much. could you help me to have a look there?

**Comment by bastijns-jeroen** [comment]
Hi @CodingFabian I only had issues with assembly where org.ops4j.io and org.ops4j.lang weren't on the classpath. So I embedded them. Url-aether already has its compile dependencies embedded. For assembly this wasn't the case.

In https://ops4j1.jira.com/browse/PAXURL-251 they added the shading in url-aether because of an issue with conflicting versions of maven and aether when url-aether is used outside of an osgi-context.

I will look at the pull-request.

**Pull request merged**